src/rest.rs

```diff
@@ -1128,6 +1128,26 @@ fn handle_request(
 
             json_response(prepare_txs(txs, query, config), TTL_SHORT)
         }
+        (&Method::POST, Some(&"mempool"), Some(&"txs"), None, None, None) => {
+            let txid_strings: Vec<String> =
+                serde_json::from_slice(&body).map_err(|err| HttpError::from(err.to_string()))?;
+
+            match txid_strings
+                .into_iter()
+                .map(|txid| Txid::from_hex(&txid))
+                .collect::<Result<Vec<Txid>, HashHexError>>()
+            {
+                Ok(txids) => {
+                    let txs: Vec<(Transaction, Option<BlockId>)> = txids
+                        .iter()
+                        .filter_map(|txid| query.mempool().lookup_txn(txid).map(|tx| (tx, None)))
+                        .collect();
+
+                    json_response(prepare_txs(txs, query, config), TTL_SHORT)
```

Comment:
I wonder if the `Cache-Control: max-age=10` header that is attached to this response is appropriate.

(Does client side JS even respect Cache-Control? I forget)